### PR TITLE
Switch to check-spelling v0.0.21

### PR DESCRIPTION
**Pull request by jsoref**
This switches check-spelling to [v0.0.21](https://github.com/check-spelling/check-spelling/releases/tag/v0.0.21) and refreshes the workflow to [check-spelling/spell-check-this@main](https://github.com/check-spelling/spell-check-this/tree/9c3ef4d7167d6dfa39d3040e763c5e7df8f6c681).

Thanks for testing the prerelease.

### CODE REVIEW

File: .github/actions/spelling/line_forbidden.patterns [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+[Pp]re[- ]emptively
+
+# s.b. reentrancy
+[Rr]e[- ]entrancy
+
+# s.b. reentrant
+[Rr]e[- ]entrant
+
+# s.b. workaround(s)
+#\bwork[- ]arounds?\b
```
Comment (jsoref) [inline .github/actions/spelling/line_forbidden.patterns:59]: Conversely, this pattern was never right and generally has a high false positive rate, so I'm turning it off.

File: .github/actions/spelling/patterns.txt [hunk 2]
```diff
@@ -1,5 +1,33 @@
...
+# IPv6
+\b(?:[0-9a-fA-F]{0,4}:){3,7}[0-9a-fA-F]{0,4}\b
+
+# hit-count: 1 file-count: 1
+# Amazon
+\bamazon\.com/[-\w]+/(?:dp/[0-9A-Z]+|)
+
+# hit-count: 1 file-count: 1
+# Non-English
+[a-zA-Z]*[ÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖØÙÚÛÜÝßàáâãäåæçèéêëìíîïðñòóôõöøùúûüýÿĀāŁłŃńŅņŒœŚśŠšŜŝŸŽžź][a-zA-Z]{3}[a-zA-ZÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖØÙÚÛÜÝßàáâãäåæçèéêëìíîïðñòóôõöøùúûüýÿĀāŁłŃńŅņŒœŚśŠšŜŝŸŽžź]*
```
Comment (jsoref) [inline .github/actions/spelling/patterns.txt:29]: Admittedly, this is a moderately expensive pattern for a single name, but check-spelling runs sufficiently quickly that it doesn't matter, and maybe someone will eventually show up w/ another name where it's valuable to have the pattern available for them too.

File: .github/actions/spelling/patterns.txt [hunk 3]
```diff
@@ -65,43 +93,46 @@ d88ff1001d4bf753a1b800001accfe22
 &U,BTFw-
 &ZeVnLIqe-
 -adnPNSx
-MTM1NDQ1WjAmMSQwIgYDjksadnjkasndjksandjksandjksandj5YXJlYS5vcmcw
--fPRU
--1fPRU
-fd95:4eed:38ba::25
+
+\bAPPEND(?:ed|s)
```
Comment (jsoref) [inline .github/actions/spelling/patterns.txt:97]: This lets me remove `APPEN` from `expect.txt`

File: .github/actions/spelling/patterns.txt [hunk 4]
```diff
@@ -65,43 +93,46 @@ d88ff1001d4bf753a1b800001accfe22
...
+
+#
+(?:Master-Key|Session-ID): [A-F0-9]+
+
+# Certificate
+^\s*[-a-zA-Z=;:/0-9+\.]{64}$
+^\s*[-a-zA-Z=;:/0-9+\.]{40,}=+$
```
Comment (jsoref) [inline .github/actions/spelling/patterns.txt:104]: These lines match the huge deleted clump below

File: .github/workflows/spelling.yml [hunk 5]
```diff
@@ -1,9 +1,56 @@
...
+    - "**"
+    tags-ignore:
+    - "**"
+    types:
+    - 'opened'
+    - 'reopened'
+    - 'synchronize'
+  issue_comment:
+    types:
+    - 'created'
```
Comment (jsoref) [inline .github/workflows/spelling.yml:53]: I didn't mean to include this (will drop) -- it's for talk-to-the-bot (as noted in the comment above)

File: source/admin_manual/debugging/debugging_rawlog.rst [hunk 6]
```diff
@@ -33,7 +33,7 @@ Pre-login rawlog
 .. versionadded:: v2.3.2
 
 The pre-login rawlog is used before IMAP, POP3, Submission or ManageSieve
-client logs in to the post-login process. Note that LMTP and doveadm protocols
+client logs into the post-login process. Note that LMTP and doveadm protocols
```
Comment (jsoref) [inline source/admin_manual/debugging/debugging_rawlog.rst:36]: As noted, `in to` / `into` has a high false-positive rate in general.

In this case, I'm fairly certain that

`The ... rawlog is used before a <client logs in> <to the something>` does not do what is intended, and the proper preposition really is:
`The ... rawlog is used before a <client logs into> <the something>`

**Review by sirainen — APPROVED** [review]

File: source/configuration_manual/acl.rst [hunk 7]
```diff
@@ -72,7 +72,7 @@ Sample Configuration:
 Master users
 ============
 
-Master users have their own ACLs. They're not the the mailbox owners, so by
+Master users have their own ACLs. They're not the mailbox owners, so by
```
Comment (sirainen) [inline source/configuration_manual/acl.rst:75]: the the changes missing:

Error: source/pigeonhole-man/sievec.1.rst.in:60:22 ... 31, Error - ` the the ` matches a line_forbidden.patterns entry: `\s([A-Z]{3,}|[A-Z][a-z]{2,}|[a-z]{3,})\s\g{-1}\s`. (forbidden-pattern)

**Review by sirainen — CHANGES_REQUESTED** [review]

File: source/configuration_manual/acl.rst (hunk 7 above)
Comment (jsoref) [reply source/configuration_manual/acl.rst:75]: Thanks, fixed in: https://github.com/dovecot/documentation/compare/731b2a8477713348594d80b71540e5462b8a3093..b1dfaae5a4908d0f11a93addbb30844977ce4348

**Review by sirainen — APPROVED** [review]

**Pull request merged**